File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,list<int>>map;
        
        for(int i=0;i<strs.size();i++){
            string str = strs[i];
            sort(str.begin(),str.end());
            map[str].push_back(i);
        }
        vector<vector<string>> ans;
        for(auto s:map){
            vector<string> grp;
            for(auto i: s.second) grp.emplace_back(strs[i]);
            ans.emplace_back(grp);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        mapping = {}

        for i, word in enumerate(strs):
            sorted_word = "".join(sorted(word))
            if sorted_word not in mapping:
                mapping[sorted_word] = []
            mapping[sorted_word].append(i)

        result = []
        for indices in mapping.values():
            group = [strs[i] for i in indices]
            result.append(group)

        return result        